src/Avalonia.Controls/ListBoxItem.cs

```diff
@@ -40,5 +45,67 @@ protected override AutomationPeer OnCreateAutomationPeer()
         {
             return new ListItemAutomationPeer(this);
         }
+
+        protected override void OnPointerPressed(PointerPressedEventArgs e)
+        {
+            base.OnPointerPressed(e);
+
+            _pointerDownPoint = s_invalidPoint;
+
+            if (e.Handled)
+                return;
+
+            if (!e.Handled && ItemsControl.ItemsControlFromItemContaner(this) is ListBox owner)
+            {
+                var p = e.GetCurrentPoint(this);
+
+                if (p.Properties.PointerUpdateKind is PointerUpdateKind.LeftButtonPressed or 
+                    PointerUpdateKind.RightButtonPressed)
+                {
+                    if (p.Pointer.Type == PointerType.Mouse)
+                    {
+                        // If the pressed point comes from a mouse, perform the selection immediately.
+                        e.Handled = owner.UpdateSelectionFromPointerEvent(this, e);
+                    }
+                    else
+                    {
+                        // Otherwise perform the selection when the pointer is released as to not
+                        // interfere with gestures.
+                        _pointerDownPoint = p.Position;
+
+                        // Ideally we'd set handled here, but that would prevent the scroll gesture
+                        // recognizer from working.
+                        ////e.Handled = true;
```

Comment:
This line should be uncommented when we fix scroll gestures to work with handled events.